### ISSUE TITLE
Cannot login into salesforces test account (not a sandbox)

### ISSUE THREAD

**Issue by tuliren** [airbytehq/airbyte#14416]
## Environment
- **Airbyte version**: latest cloud
- **Deployment**: cloud
- **Source Connector and version**: Salesforce `1.0.10`
- **Step where error happened**: Setup new connection

## Current Behavior
- Start with a Salesforce test account.
  - Pre-configured with read-only data.
  - It is not a a sandbox account.
- Without sandbox selected
  - Create a new Salesforce connector.
  - Click the "authenticate" button.
  - The oauth login url is correct.
  - After click log in, the url changes to something involving "app lock".
- With sandbox selected
  - Create a new Salesforce connector.
  - Select "sandbox".
  - Click the "authenticate" button.
  - The oauth test url shows up.
  - Cannot log in due to invalid credential.
- In summary, there is no way to authenticate properly with this account.

## Expected Behavior
- Can log into this test account.

## Steps to Reproduce
- Already listed above.
